FreeBSD does have support for utmp/utmpx, so include relevant files into
a build.

--- src/Makefile.am.orig	2019-01-29 12:51:38 UTC
+++ src/Makefile.am
@@ -8,7 +8,9 @@ tnftpd_SOURCES = \
 	ftpd.c \
 	ftpcmd.y \
 	pfilter.c \
-	popen.c
+	popen.c \
+	logutmp.c \
+	logwtmp.c
 
 noinst_HEADERS = \
 	extern.h \
@@ -51,6 +53,4 @@ seddotin = \
 EXTRA_DIST = \
 	ftpd.conf.manin \
 	ftpusers.manin \
-	logutmp.c \
-	logwtmp.c \
 	tnftpd.manin
